lang/extract_json_strings.py

```diff
@@ -583,9 +583,14 @@ def extract_monster_attack(item):
 def extract_recipes(item):
     outfile = get_outfile("recipe")
     if "book_learn" in item:
-        for arr in item["book_learn"]:
-            if len(arr) >= 3 and len(arr[2]) > 0:
-                writestr(outfile, arr[2])
+        if type(item["book_learn"]) is dict:
+            for (k, v) in item["book_learn"].items():
+                if type(v) is dict and "recipe_name" in v:
+                    writestr(outfile, v["recipe_name"])
+        else:
+            for arr in item["book_learn"]:
+                if len(arr) >= 3 and len(arr[2]) > 0:
+                    writestr(outfile, arr[2])
```

Comment:
The game cannot read things in this format anymore, you should just remove the code for it.

Reply:
Doesn't it extract that string?
```json
  {
    "type": "recipe",
    "result": "liver_onion",
    "result_mult": 2,
    "category": "CC_FOOD",
    "subcategory": "CSC_FOOD_MEAT",
    "skill_used": "cooking",
    "difficulty": 2,
    "time": "30 m",
    "batch_time_factors": [ 83, 5 ],
    "book_learn": [ [ "scots_cookbook", 2 ], [ "cookbook", 0 ], [ "cookbook_liverforkids", 1, "Candied Onions and Giblets" ] ],
    "qualities": [ { "id": "CUT", "level": 1 }, { "id": "COOK", "level": 2 } ],
    "tools": [ [ [ "surface_heat", 50, "LIST" ] ] ],
    "components": [ [ [ "liver", 8 ] ], [ [ "onion", 1 ], [ "irradiated_onion", 1 ] ] ]
  }
```
Game logic:
https://github.com/CleverRaven/Cataclysm-DDA/blob/478ced98fd31acbaef7dd6903d31572a562bbebd/src/recipe.cpp#L228-L237

Reply:
No, that array extraction code does not extract a string, it only extracts the `0` and `1` elements, the book and skill.